src/azul/chalice.py

```diff
@@ -44,11 +59,11 @@ def route(self, path, enabled=True, path_spec=None, method_spec=None, **kwargs):
                         view function wasn't decorated.
         """
         if enabled:
-            methods = kwargs.get('methods')
+            methods = kwargs.get('methods', ())
             decorator = super().route(path, **kwargs)
 
             def _decorator(view_func):
-                view_func = openapi_spec(path, methods, path_spec=path_spec, method_spec=method_spec)(view_func)
+                self._register_spec(path, path_spec, method_spec, methods)
                 # Stash the URL path a view function is bound to as an attribute of
                 # the function itself.
                 view_func.path = path
```

Comment:
Do we still need to do this?

Reply:
My changes up to this point have not removed the need for it, if that's what you're asking. I can very easily make additional changes to implement an approach that avoids storing this attribute (which AFAICT is only ever used for two specific routes in a single function).

Reply:
The ticket says

> Currently, OpenAPI spec for each route is stored as an attribute to the decorated function. A more elegant and efficient solution is to store this spec as an attribute of the Chalice app object.

which is misleading. It's not the spec that is stored in an attribute, it's the path. But the intend is to remove the attribute, whatever it stores. So please do make those changes. For future reference, and in case you were already aware of that error in the ticket description, feel free to proactively bring things like that to my attention.

Reply:
I will certainly make the changes to remove the path attribute from the functions, but I do not believe the ticket description was inaccurate. `path_spec` and `method_spec` attributes are indeed stored in the decorated functions on the develop branch. See https://github.com/DataBiosphere/azul/pull/1668/files#diff-faf2c5ae5050f2419fdbdbd6a8af190aL95

Reply:
FWIW, I didn't store the path as an attribute. It looks like this was done by @hannes-ucsc in b277a14af879a1db5c7f72862494064feab42922.

Reply:
You both are right and I was wrong. The ticket description IS accurate and the I did at the `.path` attribute. After @jessebrennan pointed out the commit, I remembered why I added it. It was for a good reason so we'll need to revert the removal. I owe you a favor, @noah-aviel-dove.